Formula/d/dotnet.rb

```diff
@@ -65,24 +66,45 @@ class Dotnet < Formula
   conflicts_with cask: "dotnet-sdk@preview"
 
   def install
+    # Fixes build error where member names shadow type names
+    # Error: declaration of '...' changes meaning of '...'
+    inreplace "src/runtime/src/coreclr/jit/gentree.h" do |s|
+      s.gsub! "    ExecutionContextHandling    ExecutionContextHandling",
+              "    ::ExecutionContextHandling    ExecutionContextHandling"
+      s.gsub! "= ExecutionContextHandling::None;",
+              "= ::ExecutionContextHandling::None;"
+
+      s.gsub! "    ContinuationContextHandling ContinuationContextHandling",
+              "    ::ContinuationContextHandling ContinuationContextHandling"
+      s.gsub! "= ContinuationContextHandling::None;",
+              "= ::ContinuationContextHandling::None;"
+    end
+
+    # Fix `unbound variable` error if an array is empty:
+    # ./prep-source-build.sh: line 254: positional_args[@]: unbound variable
+    inreplace "prep-source-build.sh", '"${positional_args[@]}"', '"${positional_args[@]:-}"'
```

Comment:
Should be reported upstream. Or try brew Bash on macOS if only an ancient Bash issue. We don't want to carry a local patch indefinitely